ktor-network/ktor-network-tls/jvm/src/io/ktor/network/tls/TLSConfigBuilder.kt

```diff
@@ -48,11 +48,20 @@ class TLSConfigBuilder {
     /**
      * Create [TLSConfig].
      */
-    fun build(): TLSConfig = TLSConfig(
-        random ?: SecureRandom.getInstanceStrong(),
-        certificates, trustManager as? X509TrustManager ?: findTrustManager(),
-        cipherSuites, serverName
-    )
+    fun build(): TLSConfig {
+        val secureRandomInstance = try {
+            SecureRandom.getInstanceStrong()
+        } catch (e: NoSuchMethodError) {
+            //SecureRandom.getInstanceStrong() requires JVM 1.8 and for Android api level 26
+            //if getInstanceStrong() is not available use non-blocking SecureRandom() to get instance
+            SecureRandom()
```

Comment:
What are the security guarantees of `SecureRandom.getInstanceStrong()` that `SecureRandom()` won't give users? Are these equivalent?

Can you include a link to some sort of official Android documentation about how to generate secure random numbers in versions of android prior to 26?

Reply:
https://tersesystems.com/blog/2015/12/17/the-right-way-to-use-securerandom/
According to this article SecureRandom.getInstanceStrong() blocks the thread if there isn't enough randomness available.. And that pretty much won't happen unless you are running on a `container based solution like Docker or CoreOS`. Other than that they are equally secure... 

https://stackoverflow.com/questions/27622625/securerandom-with-nativeprng-vs-sha1prng
[The top answer](https://stackoverflow.com/a/27638413/1363742)  advocates that specifying a provider may hurt interoperability and [links](https://android-developers.googleblog.com/2016/06/security-crypto-provider-deprecated-in.html) to an android dev post where they say `SecureRandom()` should be used

Also,
https://www.veracode.com/blog/research/cryptographically-secure-pseudo-random-number-generator-csprng#tldr

Suggests: 
> Note: This recommendation has the additional advantage of keeping code portable across operating systems, and will provide a secure randomizer if self-seeded. If you want complete assurance of randomness for a given operating system, I would suggest explicitly using the "Windows-PRNG" algorithm for Windows environments (using the getInstance method) and "NativePRNG" for Unix-like environments. Note that these options carry the downside of making code not easily portable. This is explained in detail later in this pos

Reply:
Sounds good. Thanks!